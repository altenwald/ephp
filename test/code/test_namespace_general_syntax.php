<?php
namespace my\name; // see "Defining Namespaces" section

class MyClass {
    const MYCLASSCONST = 2;
}
function myfunction() {}
const MYCONST = 1;

$a = new MyClass;
var_dump($a);
$c = new \my\name\MyClass; // see "Global Space" section
var_dump($c);

$a = strlen('hi'); // see "Using namespaces: fallback to global
                   // function/constant" section
echo "strlen = $a\n";
$d = namespace\MYCONST; // see "namespace operator and __NAMESPACE__
                        // constant" section
echo "$d\n";
$d = __NAMESPACE__ . '\MYCONST';
echo constant($d) . "\n"; // see "Namespaces and dynamic language features" section

echo namespace\MyClass::MYCLASSCONST . "\n";
echo MyClass::MYCLASSCONST . "\n";
