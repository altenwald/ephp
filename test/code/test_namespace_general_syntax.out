object(MyClass)#1 (0) {
}
object(MyClass)#2 (0) {
}
strlen = 2
1
1
2
2
